File: benchmarks/multilang-wasm/sum_u32.ts
export function sum_u32(ptr: usize, len: i32): i32 {
  let total: i32 = 0;
  for (let i: i32 = 0; i < len; i++) {
    total += load<u32>(ptr + (i << 2));
  }
  return total;
}

File: benchmarks/build-variants-c/kernels/sum_u32.c
// C source: sum_u32 kernel for build variant benchmarking.
// Compiled at different optimization levels to measure binary size and execution differences.
// Export: sum_u32(ptr: *i32, len: i32) -> i32

int sum_u32(int *ptr, int len) {
  int sum = 0;
  for (int i = 0; i < len; i++) {
    sum += ptr[i];
  }
  return sum;
}
